src/pages/groupComparison/GroupComparisonPage.tsx

```diff
@@ -506,4 +524,18 @@ export default class GroupComparisonPage extends React.Component<
             </PageLayout>
         );
     }
+    @autobind
```

Comment:
make this ```@action.bound```.  The reason is that you make multiple state mutations within and those will trigger synchronous render cycles unless you make this an action (mobx)